csrc/ir/nodes.cpp

```diff
@@ -4355,7 +4356,14 @@ std::vector<PolymorphicValue> MatmulOp::evaluate(
     const std::vector<PolymorphicValue>& inputs) const {
   const auto a = inputs.at(0).as<at::Tensor>();
   const auto b = inputs.at(1).as<at::Tensor>();
-  return {at::matmul(a, b)};
+  auto matmul_out = at::matmul(a, b);
+  if (out()->hasAllocation()) {
+    auto matmul_sizes = matmul_out.sizes().vec();
+    auto strides = computeStrides(out(), matmul_sizes);
+    matmul_out = at::as_strided(matmul_out, matmul_sizes, strides);
+  }
+  inferAndValidateAllocationSizesAndStrides(matmul_out, out(), ee);
```

Comment:
I'm not sure about validating output allocation for all MatmulOps. 
1. We already validate allocation sizes/strides for each segment's [inputs](https://github.com/NVIDIA/Fuser/blob/c154e90919c40bfe2202b432c2a38e106d1a5444/csrc/tensor_metadata.cpp#L365) and [outputs](https://github.com/NVIDIA/Fuser/blob/c154e90919c40bfe2202b432c2a38e106d1a5444/csrc/runtime/allocations.cpp#L307). Given MatmulOp currently forms its own segment, existing validation seems enough. 
2. If/When MatmulOp produces an **internal** tensor, we can't always materialize the tensor as an `at::Tensor` that matches its allocation domain. For example, the allocation domain can be a split and/or a swizzle of logical. Assuming allocation is a permutation of logical is probably OK for segment inputs/outputs, but can be too limiting for internal tensors. cc @zasdfgbnm